### ISSUE TITLE
Validation performed on test set?

### ISSUE THREAD

**Issue by andreamaracani** [thuml/Transfer-Learning-Library#105]
**Describe the bug**
In the file `examples/domain_adaptation/image_classification/utils.py` the function `get_dataset` in most cases returns `val_dataset = test_dataset `. Then, for example, in the file `examples/domain_adaptation/image_classification/mdd.py` in the training loop starting at row `113` the best model is saved, but the accuracy to decide which model is "the best" is actually computed on the validation set that is the test set.
Am I understanding it wrong or it is an unintentional bug?

**Comment by JunguangJiang** [comment]
Hope [issue15](https://github.com/thuml/Transfer-Learning-Library/issues/15), [issue28](https://github.com/thuml/Transfer-Learning-Library/issues/28) and [issue47](https://github.com/thuml/Transfer-Learning-Library/issues/47) could answer your question.

**Comment by andreamaracani** [comment]
Thank you very much for the answer, actually do you remember an example of paper that do the same?  Now I know that if I want to compare the results that I get to others I need to check where the validation is done.
Actually it is a big problem, the code of papers I used till now splits the source domain in 90%-10% (even the small Office31) and choose the iterations based on the source accuracy even if it may not be representative of the test accuracy. Actually validating on the test just the number of iterations does not increase the accuracy too much, but, even if the standard procedure in the banchmark is transductive learning and we have access to the unlabeled test set during training, we cannot use the labels of test neither at training, neither at validation: in theory everybody could get 100% accuracy in any dataset by introducing other hyperparameters and validating them on the test set. 
I underline that your code its great and I will fix this by myself, I was just complaining against the published papers that report the test-validated accuracy in their tables...

**Comment by JunguangJiang** [comment]
Since we provide the training code for many DA methods, you could run them in your setup (split the Office31 into several splits).
